File: ClientApp/src/app/core/interfaces/Product/Response/OrderManagementPreviewResponse.ts
import {AddressDto} from "@core/interfaces/Product/UserInteract/AddressDto";

export interface OrderManagementPreviewResponse {
  id: string;
  orderId: string;
  total: number;
  currencyCode: string;
  transactionDate: string | null;
  status: string;
  userId: string | null;
  address: AddressDto;
  quoteId: string | null;
}

File: Contracts/Model/Product/Response/OrderManagementPreviewResponse.cs
namespace Contracts.Model.Product.Response;

public sealed class OrderManagementPreviewResponse
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string? OrderId { get; set; }
    public double Total { get; set; }
    public string? CurrencyCode { get; set; }
    public string? TransactionDate { get; set; }
    public string Status { get; set; }
    public string? UserId { get; set; }
    public AddressDto Address { get; set; }
    public Guid QuoteId { get; set; }
}